README.md

```diff
@@ -270,49 +353,58 @@ will automatically use chunked transfer encoding and send the respective header
 If you know the length of your body, you MAY specify it like this instead:
 
 ```php
-$data = 'Hello World!';
-
-$response->writeHead(200, array(
-    'Content-Type' => 'text/plain',
-    'Content-Length' => strlen($data)
-));
-$response->end($data);
+$server = new Server($socket, function (RequestInterface $request) use ($loop) {
+    return new Response(
+        200,
+        array(
+            'Content-Type' => 'text/plain',
+        ),
+        "Hello world!\n"
+    );
+});
 ```
 
+An invalid return value or an unhandled `Exceptions` in the code of the callback
+function, will result in an `500 Internal Server Error` message.
+Make sure to catch `Exceptions` to create own response messages.
+
+After the return in the callback function the response will be processed by the `Server`.
+The `Server` will add the protocol version of the request, so you don't have to.
+
 A `Date` header will be automatically added with the system date and time if none is given.
 You can add a custom `Date` header yourself like this:
 
 ```php
-$response->writeHead(200, array(
-    'Date' => date('D, d M Y H:i:s T')
-));
+$server = new Server($socket, function (RequestInterface $request) {
+    return new Response(200, array('Date' => date('D, d M Y H:i:s T')));
+});
 ```
 
 If you don't have a appropriate clock to rely on, you should
-unset this header with an empty array:
+unset this header with an empty string:
 
 ```php
-$response->writeHead(200, array(
-    'Date' => array()
-));
+$server = new Server($socket, function (RequestInterface $request) {
+    return new Response(200, array('Date' => ''));
+});
 ```
 
 Note that it will automatically assume a `X-Powered-By: react/alpha` header
 unless your specify a custom `X-Powered-By` header yourself:
 
 ```php
-$response->writeHead(200, array(
-    'X-Powered-By' => 'PHP 3'
-));
+$server = new Server($socket, function (RequestInterface $request) {
+    return new Response(200, array('X-Powered-By' => 'PHP 3'));
```

Comment:
Well, I can still change it while I'm on it. Just used the same example as before :sweat_smile: